src/config/Config.py

```diff
@@ -29,53 +32,76 @@ def ingest_env(self):
         if "PIPELINE_DB_NAME" in os.environ:
             self._data["pipeline_db_name"] = os.environ["PIPELINE_DB_NAME"]
 
+    def _is_date(self, val):
+        if not isinstance(val, str):
+            return False
+
+        if len(val) == 10:
+            for i in range(len(val)):
+                if i == 4 or i == 7:
+                    if val[i] != '-':
+                        return False      
+                else:
+                    if not val[i].isnumeric():
+                        return False
+        else:
+            return False                
+        return True
```

Comment:
I think this is fine, though I wanna mention that you could run dateutil.parser on the string and try/except a ValueError exception to check if it's a valid date or not. This way date checking is more robust.